packages/webdriverio/src/utils/interception/devtools.js

```diff
@@ -13,13 +13,30 @@ const log = logger('webdriverio')
 export default class DevtoolsInterception extends Interception {
     static handleRequestInterception (client, mocks) {
         return async (event) => {
-            const responseHeaders = event.responseHeaders.reduce((headers, { name, value }) => {
+            // responseHeaders and responseStatusCode are only present in Response stage
+            // https://chromedevtools.github.io/devtools-protocol/tot/Fetch/#event-requestPaused
+            const isRequest = !event.responseHeaders
+            const eventResponseHeaders = event.responseHeaders || []
+            const responseHeaders = eventResponseHeaders.reduce((headers, { name, value }) => {
                 headers[name] = value
                 return headers
             }, {})
-            const { requestId, request, responseStatusCode } = event
+            const { requestId, request, responseStatusCode = 200 } = event
 
             for (const mock of mocks) {
+                /**
+                 * skip response mocks in Request stage
+                 */
+                if (isRequest && (
+                    mock.respondOverwrites.length === 0 || // nothing to do in Request stage
+                    (!mock.respondOverwrites[0].errorReason && // skip if not going to abort a request
+                    // or want to fetch response
+                    mock.respondOverwrites[0].params &&
+                    mock.respondOverwrites[0].params.fetchResponse !== false)
+                )) {
+                    continue
```

Comment:
If we continue here we don't continue the request with the mocked response we have set for the resource, no=

Reply:
Not now because of https://github.com/webdriverio/webdriverio/pull/6006/files#diff-c8a4111f05935473915df3949aab097fd9113d5daed85624222ab01bdc9eb90aR118

Listening for 2 events for each request:
1. when request is about to be sent
2. when response received (at this point we have response status code and headers, like before)

Reply:
There are two calls in the test as well https://github.com/webdriverio/webdriverio/pull/6006/files#diff-60adb4b9fe864361906b1b1169a2874bca2ec19cce0fc4dcfcfe1dac00f9b9fbR37